src/mutation_annotated_tree.cpp

```diff
@@ -1899,3 +1900,227 @@ void Mutation_Annotated_Tree::get_sample_mutation_paths (Mutation_Annotated_Tree
+                    } else {
+                        fprintf(stderr, "WARNING: Ignoring sample %s as it is already in the tree.\n", words[j].c_str());
+                    }
+                }
+                header_found = true;
+            }
+        } else if (header_found) {
+            if (words.size() != 9+variant_ids.size()) {
+                fprintf(stderr, "ERROR! Incorrect VCF format. Expected %zu columns but got %zu.\n", 9+variant_ids.size(), words.size());
+                exit(1);
+            }
+            std::vector<std::string> alleles;
+            alleles.clear();
+            string_split(words[4], ',', alleles);
+            for (size_t k = 0; k < missing_idx.size(); k++) {
+                size_t j = missing_idx[k];
+                auto iter = missing_samples.begin();
+                std::advance(iter, k);
+                if (iter != missing_samples.end()) {
+                    Mutation m;
+                    m.chrom = words[0];
+                    m.position = std::stoi(words[1]);
+                    m.ref_nuc = get_nuc_id(words[3][0]);
+                    assert((m.ref_nuc & (m.ref_nuc-1)) == 0); //check if it is power of 2
+                    m.par_nuc = m.ref_nuc;
+                    // Alleles such as '.' should be treated as missing
+                    // data. if the word is numeric, it is an index to one
+                    // of the alleles
+                    if (isdigit(words[j][0])) {
+                        int allele_id = std::stoi(words[j]);
+                        if (allele_id > 0) {
+                            std::string allele = alleles[allele_id-1];
+                            if (allele[0] == 'N') {
+                                m.is_missing = true;
+                                m.mut_nuc = get_nuc_id('N');
+                            } else {
+                                auto nuc = get_nuc_id(allele[0]);
+                                if (nuc == get_nuc_id('N')) {
+                                    m.is_missing = true;
+                                } else {
+                                    m.is_missing = false;
+                                }
+                                m.mut_nuc = nuc;
+                            }
+                            (*iter).mutations.emplace_back(m);
+                        }
+                    } else {
+                        m.is_missing = true;
+                        m.mut_nuc = get_nuc_id('N');
+                        (*iter).mutations.emplace_back(m);
+                    }
+                    if ((m.mut_nuc & (m.mut_nuc-1)) !=0) {
+                        (*iter).num_ambiguous++;
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+// If called with a tree file that needs to create a MAT
+void Mutation_Annotated_Tree::read_vcf(Mutation_Annotated_Tree::Tree* T, std::string &vcf_filename, std::vector<Missing_Sample>& missing_samples, bool create_new_mat) {
```

Comment:
What's the point of the boolean `create_new_mat`? It doesn't seem like you're using it inside the function?